Add kafka exporter (#1947)
strimzi/strimzi-kafka-operator: documentation/book/appendix_crds.adoc

|xref:type-CertificateAuthority-{context}[`CertificateAuthority`]
|clientsCa               1.2+<.<|Configuration of the clients certificate authority.
|xref:type-CertificateAuthority-{context}[`CertificateAuthority`]

> |kafkaExporter           1.2+<.<|Configuration of the Kafka Exporter. Kafka Exporter can gain Kafka specific metrics, for example lag of consumer group at topic/partition.

Reviewer: ```suggestion
|kafkaExporter           1.2+<.<|Configuration of the Kafka Exporter. Kafka Exporter can provide Kafka-specific metrics, such as consumer lag for a topic/partition.
```

|xref:type-KafkaExporterSpec-{context}[`KafkaExporterSpec`]

> |maintenanceTimeWindows  1.2+<.<|A list of time windows for the maintenance tasks (that is, certificates renewal). Each time window is defined by a cron expression.

Reviewer: ```suggestion
|maintenanceTimeWindows  1.2+<.<|A list of time windows for maintenance tasks (that is, certificate renewal). Each time window is defined by a cron expression.
```

|string (one of [replace-key, renew-certificate])
|====

[id='type-KafkaExporterSpec-{context}']
### `KafkaExporterSpec` schema reference

Used in: xref:type-KafkaSpec-{context}[`KafkaSpec`]

[options="header"]
|====
|Property                    |Description
|image                1.2+<.<|The docker image for the pods.
|string

> |groupRegex           1.2+<.<|Regex that determines which consumer groups to collect.

Reviewer: ```suggestion
|groupRegex           1.2+<.<|Regular expression to specify consumer groups to include in the metrics..
```

|string

> |topicRegex           1.2+<.<|Regex that determines which topics to collect.

Reviewer: ```suggestion
|topicRegex           1.2+<.<|Regular expression to specify the topics to include in the metrics.
```

|string

> |resources            1.2+<.<|Resource constraints (limits and requests).

Reviewer: ```suggestion
|resources            1.2+<.<|CPU and memory resources to reserve (limits and requests).
```

|xref:type-ResourceRequirements-{context}[`ResourceRequirements`]

> |logging              1.2+<.<|Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal].

Reviewer: Can you specify OFF?

|string

> |enableSaramaLogging  1.2+<.<|Turn on Sarama logging.

Reviewer: ```suggestion
|enableSaramaLogging  1.2+<.<|Enable Sarama logging, , a Go client library used by the Kafka Exporter.
```

> |enableSaramaLogging  1.2+<.<|Turn on Sarama logging.

Reviewer: Would we ever want this disabled?

Reviewer 2: Yes, because it might create a lot of undesired noise.

|boolean

> |template             1.2+<.<|Template for Kafka Exporter resources. The template allows users to specify how is the `Deployment` and `Pods` generated.

Reviewer: ```suggestion
|template             1.2+<.<|Customization of deployment templates and pods.
```

|xref:type-KafkaExporterTemplate-{context}[`KafkaExporterTemplate`]

> |livenessProbe        1.2+<.<|Pod liveness checking.

Reviewer: ```suggestion
|livenessProbe        1.2+<.<|Pod liveness check.
```

|xref:type-Probe-{context}[`Probe`]

> |readinessProbe       1.2+<.<|Pod readiness checking.

Reviewer: ```suggestion
|readinessProbe       1.2+<.<|Pod readiness check.
```
